pallets/amm/src/lib.rs

```diff
@@ -399,13 +401,20 @@ decl_module! {
 
 			Self::deposit_event(RawEvent::RemoveLiquidity(who.clone(), asset_a, asset_b, liquidity_amount));
 
-			if liquidity_left == 0 {
+			ensure!(
```

Comment:
This check MUST be done before any transfers. You have changed the state here and after the fact failing the transaction but with already changed state.

Reply:
You'll need to check if balance of share tokens of the holder after the trade is higher than minimal limit before doing that operation.